pkg/virt-controller/watch/vm.go

```diff
@@ -336,15 +340,34 @@ func (c *VMController) execute(key string) error {
 	return syncErr
 }
 
-func (c *VMController) authorizeDataVolume(vm *virtv1.VirtualMachine, dataVolume *cdiv1.DataVolume) error {
-	if dataVolume.Spec.SourceRef != nil {
+func (c *VMController) handleCloneDataVolume(vm *virtv1.VirtualMachine, dv *cdiv1.DataVolume) error {
+	if dv.Spec.SourceRef != nil {
 		return fmt.Errorf("DataVolume sourceRef not supported")
 	}
 
-	if dataVolume.Spec.Source == nil || dataVolume.Spec.Source.PVC == nil {
+	if dv.Spec.Source == nil || dv.Spec.Source.PVC == nil {
 		return nil
 	}
 
+	// For consistency with other k8s objects, we allow creating clone DataVolumes even when the source PVC doesn't exist.
+	// This means that a VirtualMachine can be successfully created with volumes that may remain unpopulated until the source PVC is created.
+	// For this reason, we check if the source PVC exists and, if not, we trigger an event to let users know of this behavior.
+	_, err := c.clientset.CoreV1().PersistentVolumeClaims(dv.Spec.Source.PVC.Namespace).Get(context.Background(), dv.Spec.Source.PVC.Name, metav1.GetOptions{})
```

Comment:
You should use the pvcInformer here, take a look at `GetPersistentVolumeClaimFromCache()`

Reply:
@mhenriks Thanks for the suggestion. I'm really struggling to write unit-tests for the no-source PVC case after changing to `GetPersistentVolumeClaimFromCache`. The reactor doesn't work, do you have any idea?

Reply:
maybe you have to add the pvc to the informer cache?  This is done in a bunch of places in vm_test.go.  Look there for examples where `pvcInformer` is used